request.go

```diff
@@ -28,6 +28,9 @@ import (
 	"github.com/notaryproject/tspclient-go/internal/oid"
 )
 
+// nonceGenerator generates nonce. Used for unit test
+var nonceGenerator func() (*big.Int, error) = generateNonce
```

Comment:
Try not using this pattern to inject unit tests.

Reply:
Think about hooking the `rand.Reader`.

Reply:
@shizhMSFT We have a similar pattern in notation-go though: https://github.com/notaryproject/notation-go/blob/82014a953fb9ccd44330eb922a5fe5b613d66237/dir/path.go#L91

Reply:
I see. Then we need to refactor the corresponding notation-go design as well. Will create an issue to track it.

Reply:
Updated.